linux-tools/libndp/libndp.py

```diff
@@ -0,0 +1,168 @@
+#!/bin/python
+import aexpect
+from distutils.spawn import find_executable
+from autotest.client import test, utils
+from autotest.client.shared import error
+
+
+class libndp(test.test):
+
+    """
+    libndp -Library for Neighbor Discovery Protocol
+
+    Neighbor Discovery Protocol (NDP) is a part of the Internet Protocol suite used with IPv6.
+    NDP Protocol make use of ICMPv6 messages and solicited-node multicast addresses for operating    its core function, which is tracking and discovering other IPv6 hosts that are present on the    other side of connected interfaces.
+
+    libndp package contains a library,libndp.so :which provides a wrapper for IPv6 Neighbor         Discovery Protocol and ndptool :tool for sending and receiving NDP messages.
+
+    @author Ramya BS <ramya@linux.vnet.ibm.com>
+
+    """
+    version = 2
+    nfail = 0
+    interface = utils.subprocess.Popen(
+        ["ip", "-6", "route", "show", "default"], stdout=utils.subprocess.PIPE)
+    interface_out = interface.stdout.read()
+    get_active_interface = utils.re.search(
+        "default via\s+\S+\s+dev\s+(\w+)\s+proto", interface_out)
+    iface = get_active_interface.group(1)
+    get_ipv6_addr = utils.subprocess.Popen(
+        ["ip", "addr", "show", "dev", "eth0"], stdout=utils.subprocess.PIPE)
+    get_ipv6_addr_out = get_ipv6_addr.stdout.read()
+    ipv6_addr = utils.re.search(
+        "inet6\s+(fe80.*)\s+scope",
+        get_ipv6_addr_out).group(1).strip("/64")
+    ns = "Type: NS"
+    rs = "Type: RA"
+
+    def install_check(self):
+        """
+        install check for libndp library and ndptool binary
+        check for ip ,route ,tcpdump
+        """
+
+        if self.iface is None:
```

Comment:
self.iface is set globally ,since will be using it in more than one method. yes you are  correct .